AsyncDisplayKit/ASCellNode.m

```diff
@@ -32,6 +40,35 @@ - (instancetype)init
   return self;
 }
 
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock
+{
+  return [self initWithViewControllerBlock:viewControllerBlock didLoadBlock:nil];
+}
+
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock didLoadBlock:(ASDisplayNodeDidLoadBlock)didLoadBlock
+{
+  if (!(self = [super init]))
+    return nil;
+  
+  ASDisplayNodeAssertNotNil(viewControllerBlock, @"should initialize with a valid block that returns a UIViewController");
+  
+  _viewController = viewControllerBlock();
+  
+  _viewControllerNode = [[ASDisplayNode alloc] initWithViewBlock:^UIView *{
+        return _viewController.view;
```

Comment:
I am not certain, but I think this probably creates a retain cycle.  It's probably reasonable to put "return viewController.view;" without accessing the instance variable, oddly - then the block will capture the view controller directly, rather than capturing self so that it can access specifically the value of that instance variable at the time the block is run (rather than just the current value of viewController, the argument)